panels/updates/cc-updates-panel.c

```diff
@@ -0,0 +1,684 @@
+    }
+
+  /* Icon */
+  icon_name = get_network_status_icon_name (self, device, ap);
+  gtk_image_set_from_icon_name (GTK_IMAGE (self->network_status_icon), icon_name, 4);
+
+  /* Name */
+  if (ap)
+    {
+      GBytes *ssid = nm_access_point_get_ssid (ap);
+      g_autofree gchar *title = nm_utils_ssid_to_utf8 (g_bytes_get_data (ssid, NULL), g_bytes_get_size (ssid));
+
+      gtk_label_set_label (GTK_LABEL (self->network_name_label), title);
+    }
+  else
+    {
+      if (connection)
+        gtk_label_set_label (GTK_LABEL (self->network_name_label), _("Connected"));
+      else
+        gtk_label_set_label (GTK_LABEL (self->network_name_label), _("No active connection"));
+    }
+
+  /* Metered status */
+  gtk_widget_set_visible (self->metered_data_label, connection != NULL);
+
+  if (connection)
+    {
+      NMSettingConnection *setting;
+      NMMetered metered;
+
+      setting = nm_connection_get_setting_connection (connection);
+      metered = nm_setting_connection_get_metered (setting);
+
+      if (metered == NM_METERED_YES || metered == NM_METERED_GUESS_YES)
+        gtk_label_set_label (GTK_LABEL (self->metered_data_label), _("Limited data plan connection"));
+      else
+        gtk_label_set_label (GTK_LABEL (self->metered_data_label), _("Unlimited data plan connection"));
+    }
+
+  /* Automatic Updates switch */
+  gtk_widget_set_sensitive (self->automatic_updates_container, connection != NULL);
+
+  ensure_setting_user (self, connection);
+
+  if (connection)
+    {
+      NMSettingUser *setting_user;
+      const gchar *value;
+
+      setting_user = NM_SETTING_USER (nm_connection_get_setting (connection, NM_TYPE_SETTING_USER));
+      value = nm_setting_user_get_data (setting_user, NM_SETTING_ALLOW_DOWNLOADS_WHEN_METERED);
+
+      g_signal_handlers_block_by_func (self->automatic_updates_switch,
+                                       on_automatic_updates_switch_changed_cb,
+                                       self);
+
+      gtk_switch_set_active (GTK_SWITCH (self->automatic_updates_switch), g_strcmp0 (value, "1") == 0);
+
+      g_signal_handlers_unblock_by_func (self->automatic_updates_switch,
+                                         on_automatic_updates_switch_changed_cb,
+                                         self);
+    }
+}
+
+static void
+load_tariff (CcUpdatesPanel *self)
+{
+  g_autoptr(MwtTariffLoader) tariff_loader = NULL;
+  g_autoptr(GBytes) bytes = NULL;
+  g_autoptr(GError) error = NULL;
+  g_autofree gchar *tariff_path = NULL;
+  gchar *contents = NULL;
+  gsize size = 0;
+
+  g_debug ("Loading tariff");
+
+  tariff_path = g_build_filename (g_get_user_config_dir (), SYSTEM_TARIFF_FILENAME, NULL);
+  g_file_get_contents (tariff_path, &contents, &size, &error);
+  if (error)
+    {
+      if (!g_error_matches (error, G_FILE_ERROR, G_FILE_ERROR_NOENT))
+        g_warning ("Error reading tariff file: %s", error->message);
+      return;
+    }
+
+  bytes = g_bytes_new_take (contents, size);
```

Comment:
…and use `g_steal_pointer()` here. I realise this makes no functional difference, but it does make the ownership transfer more obvious, and means that refactoring the code is less likely to cause a double-free or leak.